tpr/tpr.go

```diff
@@ -22,48 +28,74 @@ const (
 // Config is a TPR configuration.
 type Config struct {
 	// Dependencies.
-	Clientset kubernetes.Interface
+	K8sClient            kubernetes.Interface
+	Logger               micrologger.Logger
+	ResourceEventHandler cache.ResourceEventHandler
+	ZeroObjectFactory    ZeroObjectFactory
```

Comment:
I don't like this. This is why I created the informer initially as separate package, because it feels like we are overloading the TPR package. I will add the event handler and the object factory as runtime arguments to `NewInformer`, just to get this done now. Takes way longer than I wanted.